app/src/main/java/io/gnosis/kouban/ui/address/capture/AddressCaptureViewModel.kt

```diff
@@ -5,39 +5,49 @@ import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import io.gnosis.kouban.core.managers.SafeAddressManager
 import io.gnosis.kouban.core.ui.base.Error
+import io.gnosis.kouban.core.ui.base.Loading
 import io.gnosis.kouban.core.ui.base.ViewState
+import io.gnosis.kouban.data.repositories.SafeRepository
 import io.gnosis.kouban.push.PushServiceRepository
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
 import pm.gnosis.model.Solidity
 
 class AddressCaptureViewModel(
+    private val safeRepository: SafeRepository,
     private val safeAddressManager: SafeAddressManager,
     private val pushRepo: PushServiceRepository
-    ) : ViewModel() {
+) : ViewModel() {
 
     val safeAddressEvents = MutableLiveData<ViewState>()
 
     fun handleSafeAddress(safeAddress: Solidity.Address) {
         viewModelScope.launch(Dispatchers.IO) {
-            safeAddressManager.storeSafeAddress(safeAddress)
-            pushRepo.checkRegistration()
-            safeAddressEvents.postValue(SafeAddressUpdated(safeAddress))
+            runCatching { safeRepository.checkSafe(safeAddress) }
+                .onSuccess { (masterCopyAddress, isValid) ->
+                    if (masterCopyAddress != null && isValid && SafeRepository.isSupported(masterCopyAddress)) {
```

Comment:
`isValid` is a very wrong name for what is returned here. `is2FaEnabled` would be the more correct boolean and with this (as mentioned below) this logic doesn't make sense.

Also the whole 2FA check was mainly from the time when everything was happening via the mobile app. There we had a pretty static setup and new that when a safe has 3 owners with threshold 1 it does not have a 2fa option enabled and if it has 4 owners with threshold 2 it has a 2fa option enabled. 

Bit since this prototype is for all Safes (actually primarily for the ones created by the web interface) this is does not apply here